modules/renter/hostdb/hosttree/hosttree.go

```diff
@@ -181,6 +185,24 @@ func (n *node) remove() {
 	}
 }
 
+// All returns all of the hosts in the host tree, sorted by weight.
+func (ht *HostTree) All() []modules.HostDBEntry {
+	ht.mu.Lock()
+	defer ht.mu.Unlock()
+
+	var he hostEntries
+	for _, node := range ht.hosts {
+		he = append(he, *node.entry)
+	}
+	sort.Sort(he)
```

Comment:
you are thinking of `sort.Slice`. I'm just suggesting renaming `hostEntries` to `byWeight`.